### ISSUE TITLE
Using mel spectrogram in different vocoders

### ISSUE THREAD

**Issue by gafsd** [espnet/espnet#2998]
Hello I am using ESPNet2 and trying to feed the mel spectrogram I get from FS2 into different vocoders (not PWG). But the output of the spectrogram is not going into the other vocoders properly even if I use the right fmax and fmin. Is there some denormalisation I need to do? I have tried using both the spectrogram and the denormalised spectrogram I get from calling the text2speech model

**Comment by kan-bayashi** [comment]
Maybe the base of log is different.
This thread will help you.
https://github.com/kan-bayashi/ParallelWaveGAN/issues/169#issuecomment-649320778

**Comment by gafsd** [comment]
I tried this but not working
```
xx, c, c_d, *_ = text2speech(text=args.text) 
    decompressed_log10 = torch.log10(torch.exp(c)).cuda()
    mu = torch.from_numpy(read_hdf5("stats.h5", "mean")).cuda()
    sigma = torch.from_numpy(np.sqrt(read_hdf5("stats.h5", "scale"))).cuda()
    decompressed_log10_norm = (torch.transpose(decompressed_log10, 0, 1) - mu.view(1, -1, 1)) / sigma.view(1, -1, 1) 
    audio = model.inference(decompressed_log10_norm)
```

**Comment by kan-bayashi** [comment]
c is normalized one.
You should use denormalized one to normalize with another stats.
And the base of log is 10.
See 
https://github.com/espnet/espnet/blob/6e1c4dd5fb08f2d8d838cf631ac4dcc7c3f11b1b/espnet2/tts/feats_extract/log_mel_fbank.py#L68

**Comment by gafsd** [comment]
Ok, I tried with denormalized one but not work and I used log10 in code?

**Comment by kan-bayashi** [comment]
We use log10 but maybe the other impl use ln.

**Comment by kan-bayashi** [comment]
Please carefully check both codes.

**Comment by gafsd** [comment]
Hm, seems to use magnitudes direct at least in this one?
https://github.com/rishikksh20/TFGAN/blob/main/utils/stft.py#L154
https://github.com/rishikksh20/TFGAN/blob/master/model/generator.py#L62

**Comment by kan-bayashi** [comment]
Please check your code again.
We use log10 and want to change to ln.
But you used exp at first.

**Comment by kan-bayashi** [comment]
The shared comment use the other repo txt2mel and my PWG.
Your case is my txt2mel and the other impl vocoder.

**Comment by gafsd** [comment]
Yes, sorry, just noticed. So just checking: c_d is in log10?

**Comment by kan-bayashi** [comment]
Yes, as you saw in https://github.com/espnet/espnet/blob/6e1c4dd5fb08f2d8d838cf631ac4dcc7c3f11b1b/espnet2/tts/feats_extract/log_mel_fbank.py#L68

**Comment by gafsd** [comment]
Hmmm it sound bad still but not sure if it is code or vocoder training... have tried changing to various bases 
```
xx, c, c_d, *_ = text2speech(text=args.text) 
    lg10 = c_d.cuda()
    mu = torch.from_numpy(read_hdf5("stats.h5", "mean")).cuda()
    sigma = torch.from_numpy(np.sqrt(read_hdf5("stats.h5", "scale"))).cuda()
    lg10_norm = (torch.transpose(lg10, 0, 1) - mu.view(1, -1, 1)) / sigma.view(1, -1, 1)  
    audio = tfgan.inference(torch.log(torch.pow(lg10_norm, 10)))
```

**Comment by kan-bayashi** [comment]
Is this stats vocoder’s one?
If so, why did not you change the base at first before normalization?

**Comment by kan-bayashi** [comment]
You should carefully check the order of the each operation.

**Comment by gafsd** [comment]
Yes stats from PWG dataset (same data). I thought ESPnet2 c_d [base 10] -> cmvn [base 10] -> switch to ln 
I will check again

**Comment by kan-bayashi** [comment]
Well, it is difficult to understand your procedure...
If the vocoder did not perform the normalization, you do not need to normalize again.

**Comment by gafsd** [comment]
Yes, I agree. But I tried doing with normalized before 
```
xx, c, c_d, *_ = text2speech(text=args.text)  
    audio = tfgan.inference(torch.log(torch.pow(torch.transpose(c.cuda(), 0, 1), 10)).unsqueeze(0))
```
gives me https://www.dropbox.com/s/yw7a0kkm18xrghn/out.wav?dl=1
nearly can make out sound but seems wrong

**Comment by kan-bayashi** [comment]
Why did you use c?

**Comment by gafsd** [comment]
Both c and c_d sound mostly the same. I tried both.

**Comment by kan-bayashi** [comment]
I give up today. Please find out by yourself.
If the melspc impl is the same as https://github.com/kan-bayashi/ParallelWaveGAN/issues/169#issuecomment-649320778, you can do it theoretically.

**Comment by gafsd** [comment]
Ok, thank you very much for trying to help. I will keep looking. It must be possible theoretically since all these vocoders accepts mel input.
